homework/index.js

```diff
@@ -29,18 +14,99 @@
     });
     return elem;
   }
+  function createRepositoryDescription(repoContainer, repository) {
+    repoContainer.innerHTML = '';
+    const table = createAndAppend('table', repoContainer);
+    const tBody = createAndAppend('tbody', table);
+    const details = ['Repository:', 'Description:', 'Forks:', 'Updated:'];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tBody);
+      createAndAppend('td', tr, { class: 'label', text: detail });
+      createAndAppend('td', tr, { id: detail });
+    });
+    const secondTd = document.getElementById('Repository:');
+    const link = createAndAppend('a', secondTd, {
+      href: repository.html_url,
+      target: '_blank',
+    });
+    link.innerText = repository.name;
+    document.getElementById('Description:').innerText = repository.description;
+    document.getElementById('Forks:').innerText = repository.forks;
+    document.getElementById('Updated:').innerText = new Date(
+      repository.updated_at,
+    ).toLocaleDateString();
+  }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function createContributorsSide(contributorContainer, contributors) {
+    contributorContainer.innerHTML = '';
+    createAndAppend('p', contributorContainer, { text: 'Contributions: ' });
+    contributors.forEach(contributor => {
+      const contributorInfo = createAndAppend('a', contributorContainer, {
+        href: contributor.html_url,
+        target: '_blank',
+      });
+      const contributorDiv = createAndAppend('div', contributorInfo, {
+        class: 'contributor',
+      });
+      createAndAppend('img', contributorDiv, { src: contributor.avatar_url });
+      createAndAppend('p', contributorDiv, { text: contributor.login });
+      createAndAppend('div', contributorDiv, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
     });
   }
 
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('div', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+
+    fetch(url)
+      .then(Response => Response.json())
```

Comment:
Hi Yaseir, you should try to keep your variable names in lowercase or camelCase. Pascal casing names like `Response` in this case are usually used for naming classes or constructor functions. 🙂